### PR TITLE
Allow dotenv files from different environments for SSR and SSG in Next.js

### PR DESCRIPTION
**Pull request by chalosalvador**
### Description

This PR is intended to fix the issue #7207 by checking if the .env.<PROJECT-ID> file exists and make it available for the build process.

### Scenarios Tested

1. Initialized a hosting project with Next.js
2. Created a `.env.<PROJECT-ID>` file in the root of the Next.js project
3. Created a page that uses SSR and accesses the environment variable in `.env.<PROJECT-ID>`
4. Created a page that uses SSG and accesses the environment variable in `.env.<PROJECT-ID>`
5. Deployed the project to Firebase hosting. 
6. Make sure the environment variable in the `.env.<PROJECT-ID>` is available in a server side rendered page.
7. Make sure the environment variable in the `.env.<PROJECT-ID>` is available in a static generated page.

### CODE REVIEW

File: src/frameworks/next/index.ts [hunk 1]
```diff
@@ -126,6 +126,22 @@ export async function build(
...
   const env = { ...process.env };
 
+  // Check if the .env.<PROJECT-ID> file exists and make it available for the build process
+  if (context?.projectId) {
+    const envFile = join(dir, `.env.${context.projectId}`);
+
+    if (await pathExists(envFile)) {
+      const envConfig = await readFile(envFile);
+      const envConfigString = envConfig.toString();
+      const envConfigLines = envConfigString.split("\n");
```
Comment (jamesdaniels) [inline src/frameworks/next/index.ts:136]: I worry this is a bit naive of a parser for .env, we need to make sure that we handle escaping and multiline files... maybe we should see if there's some reputable NPM libs?

**Review by jamesdaniels — CHANGES_REQUESTED** [review]
Can we add some test coverage?

File: src/frameworks/next/index.ts (hunk 1 above)
Comment (chalosalvador) [reply src/frameworks/next/index.ts]: It turns out there's actually a function available ([parseStrict](https://github.com/firebase/firebase-tools/blob/chalosalvador/dotenv-files/src/functions/env.ts#L192)) that parses .env files in a more accurate way.

File: src/frameworks/next/index.ts [hunk 2]
```diff
@@ -126,6 +127,18 @@ export async function build(
...
 
+  // Check if the .env.<PROJECT-ID> file exists and make it available for the build process
+  if (context?.projectId) {
+    const projectEnvPath = join(dir, `.env.${context.projectId}`);
+
+    if (await pathExists(projectEnvPath)) {
+      const projectEnvVars = parseStrict((await readFile(projectEnvPath)).toString());
+
+      // Merge the parsed variables with the existing environment variables
+      Object.assign(env, projectEnvVars);
```
Comment (jamesdaniels) [inline src/frameworks/next/index.ts:138]: dotenv does not override environment variables by default if i recall, so this logic will likely need inverting `{ ...projectEnvVars, ...process.env }`

**Review by jamesdaniels — CHANGES_REQUESTED** [review]
See small comment on overrides, add tests, and changelog—otherwise LGTM

File: src/frameworks/next/index.ts (hunk 2 above)
Comment (chalosalvador) [reply src/frameworks/next/index.ts]: @jamesdaniels I just made this change, added tests for the `parseStrict` function and added the changelog.

**Review by jamesdaniels — APPROVED** [review]

**Pull request merged**